test/studies/shootout/chameneos-redux/bradc/chameneosredux-blc.chpl

```diff
@@ -36,82 +53,35 @@ proc printColorEquations() {
 
 
 //
-// Given a number of chameneos as input, create a population of that
-// size, have it print its colors, host 'n' meetings, and print notes
-// about the meetings.
+// Hold meetings among the population by creating a shared meeting
+// place, and then creating per-chameneos tasks to have meetings.
 //
-proc simulate(numChameneos) {
-  const group = new Population(numChameneos);
-
-  group.printColors();
-  group.holdMeetings(n);
-  group.printNotes();
-}
+proc holdMeetings(chameneos, numMeetings) {
+  const place = new MeetingPlace(numMeetings);
 
+  coforall c in chameneos do           // create a task per chameneos
+    c.haveMeetings(place, chameneos);
 
-//
-// special colors to use for a chameneos population of size 10
-//
-const colors10 = [blue, red, yellow, red, yellow, blue, red, yellow, red, blue];
+  delete place;
+}
 
 //
-// a chameneos population
+// Print the chameneos' initial colors, the number of meetings each
+// had, and the number of self-meetings each had (spelled out).
+// Then spell out the total number of meetings for the population
 //
-record Population {
-  const size = 0;   // the size of the population
-
-  //
-  // an array of chameneos objects representing the population
-  //
-  var chameneos = [i in 1..size]
-                    new Chameneos(i, if size == 10 then colors10[i]
-                                                   else ((i-1)%3): Color);
-
-  //
-  // Print the colors of the current population.
-  //
-  proc printColors() {
-    for c in chameneos do
-      write(" ", c.color);
-    writeln();
-  }
-
-  //
-  // Hold meetings among the population by creating a shared meeting
-  // place, and then creating per-chameneos tasks to have meetings.
-  //
-  proc holdMeetings(numMeetings) {
-    const place = new MeetingPlace(numMeetings);
-
-    coforall c in chameneos do           // create a task per chameneos
-      c.haveMeetings(place, chameneos);
-
-    delete place;
+proc print(chameneos) {
```

Comment:
arg name `chameneos` confused me. What do you think about `chameneosArr` or `chameneosGroup` or something?